src/renderer.ts

```diff
@@ -96,17 +97,69 @@ export class Renderer {
             beamStart = -1;
           }
           break;
+        case 'partStart':
+          curPart = message.id;
+          if (yCur.get(`${curPart}`) == undefined) {
+            yCur.set(`${curPart}`, 100 * message.ndx);
+            factory.getContext().resize(2000, 100 * message.ndx + 300);
+            t.literal(`factory.getContext().resize (2000, ${100 * message.ndx + 300});`);
+          }
+          break;
+        case 'partEnd':
+          if (message.from == message.ndx) {
+            curMeasure++;
+            t.expression(() => (voicesArr = []));
+            t.expression(() =>
+              voices.forEach((voice) => {
+                voicesArr.push(voice);
+              })
+            );
+            t.expression(() => (formatter = factory.Formatter().joinVoices(voicesArr)));
+            const staveWidth = width
+              ? width
+              : formatter!.preCalculateMinTotalWidth(voicesArr) +
+                staves.get(`${curPart}_1`)!.getNoteStartX() -
+                staves.get(`${curPart}_1`)!.getX() +
+                VF.Stave.defaultPadding;
+            staves.forEach((stave) => {
+              stave.setWidth(staveWidth);
+            });
+            t.literal(`staves.forEach((stave) => {stave.setWidth(${staveWidth});});`);
+            const notesWidth =
+              staveWidth -
+              staves.get(`${curPart}_1`)!.getNoteStartX() +
+              staves.get(`${curPart}_1`)!.getX() -
+              VF.Stave.defaultPadding;
+            formatter!.format(voicesArr, notesWidth);
+            t.literal(`formatter.format(voicesArr, ${notesWidth});`);
+            t.expression(() => factory.draw());
+            x += staves.get(`${curPart}_1`)!.getWidth();
+            //t.expression(() => voices.clear());
+            //t.expression(() => staves.clear());
+            if (x > 1500) {
```

Comment:
It seems like we need to formalize what the context's width should be. Later on, I think we need to let the renderer know what width to use. For now, do you mind lifting all the values like this into named constants so we can see the relationship between all of them more clearly? Feel free to do that in a downstream PR.

Reply:
At want stage I will add support for print messages. MusicXML dictates the width of the page.

Reply:
SG